src/lib/coda_base/transaction_logic.ml

```diff
@@ -420,11 +420,14 @@ module Make (L : Ledger_intf) : S with type ledger := L.t = struct
     in
     match User_command.Payload.body payload with
     | Stake_delegation (Set_delegate {new_delegate}) ->
+        (*new delegate should be in the ledger*)
```

Comment:
Hm, generally we've leaned towards "all payments create the account on-demand, destination doesn't need to exist in the ledger", which this changes. Is this a conscious shift? Should fee transfers etc also require the account to be present in the ledger?

Reply:
I was confused, this was just for setting a delegate. Do we need to check this in the snark? I think we don't need to pay for any extra merkle lookups?

Reply:
I think we don't want to force the account to be present -- example use case: I send a transaction to fund some account and delegate stake to it immediately before the first transaction goes through. I should still be able to delegate to that account.

Reply:
We don't create the delegate account (or I couldn't find where it is created) but retrieve it during vrf evaluation. So vrf evaluations could potentially fail?
For all other transactions we create if the accounts are not in the ledger (except sender account in user commands)

Reply:
@bkase  The second transaction wouldn't go through if the first one doesn't because the nonce wouldn't match. If I send the delegation transaction before sending the funds or someone else sends the funds first and the fund transfer fails for any reason the account wouldn't be in the ledger.

Reply:
@emberian  we probably don't have to check this in the snark. This would be more of a sanity check, I guess

Reply:
We should probably check this in the transaction snark, I think? We wouldn't want to allow someone to delegate their stake mistakenly to an account that doesn't exist since that would effectively take that stake offline until they issue another delegation transaction.